pkg/pluginversion/pluginversion.go

```diff
@@ -37,13 +37,21 @@ func CheckPluginExists(project string, version string) (bool, error) {
 	// FIXME: would be nice if we could use `HEAD` here
 	resp, err := http.Get(pluginUrl)
 
-	if resp.StatusCode == http.StatusOK {
-		return true, nil
+	// We got an error, bail
+	if err != nil {
+		return false, err
 	}
+
+	// Not found, there's no plugin yet
 	if resp.StatusCode == http.StatusNotFound {
```

Comment:
I think this can be rewritten to check for 2xx status code and return true and then return false for every other status code - no need for the 2 If statements?

Reply:
Unfortunately not because for not found, we explicitly want to return "false" and nil for an error, so the loop continues. Every other case is probably a valid error (5xx) for example? Thoughts?